### PR TITLE
docs: Adjust developer-notes.md to use signed ints normally

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
I actually learned more about this when I went to try to convert some ints that should never be negative into uints. However, I researched it a bit and found out it's actually a quite bad idea due to what happens when signed integers and unsigned ints get mixed up. I think it'd be valuable to formalize and put into developer docs.

(also change some unsigned ints to ints that were used in arithmetic with signed ints)

### CODE REVIEW

**Comment by xdustinface** [comment]
> and fixed linter

looks still unhappy?

**Comment by PastaPastaPasta** [comment]
Okay, hopefully that makes it happy

File: doc/developer-notes.md [hunk 1]
```diff
@@ -407,6 +407,14 @@ class A
 
   - *Rationale*: Easier to understand what is happening, thus easier to spot mistakes, even for those
   that are not language lawyers
+
```
Comment (UdjinM6) [inline doc/developer-notes.md:410]:
```suggestion

```

**Review by UdjinM6 — APPROVED** [review]
ACK

**Review by xdustinface — APPROVED** [review]
utACK

**Pull request merged**